### DISCUSSION TITLE
Installation Error

### DISCUSSION THREAD

**Discussion by p0intbl4nc** [projectdiscovery/nuclei#1359] [Q&A] [answered]
# I'm unable to install nuclei. I'm getting the following error.

github.com/syndtr/goleveldb/leveldb/filter
go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/filter/bloom.go:86:13: internal compiler error: '(*bloomFilterGenerator).Generate': panic during lower while compiling (*bloomFilterGenerator).Generate:

runtime error: index out of range [1] with length 0
## I'm using the latest version of go - go1.17.3

![Screenshot_2021-12-03_10_36_38](https://user-images.githubusercontent.com/48901974/144629894-0bc32770-ca20-40fb-a0e8-563774f73262.png)

**Comment by Mzack9999** [comment]
@sreeharisj23 Thanks for opening this issue. Would it be possible to provide the following information so that we can properly try to reproduce the problem:
- Platform (e.g., the output of `uname -a`)
- Go version (e.g., the output of `go env`)
- Command used to generate the bug (I suppose `go install github.com/projectdiscovery/nuclei@latest`)?

Thanks!

**Reply by p0intbl4nc** [reply]
Platform (e.g., the output of uname -a)
Linux kali 5.14.0-kali4-amd64 #1 SMP Debian 5.14.16-1kali1 (2021-11-05) x86_64 GNU/Linux

Go version (e.g., the output of go env)
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/kali/.cache/go-build"
GOENV="/home/kali/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/kali/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/kali/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go-1.17"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go-1.17/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.17.3"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build82584554=/tmp/go-build -gno-record-gcc-switches"

Command used to generate the bug 
go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@latest

**Comment by p0intbl4nc** [comment, marked as answer]
I previously had the error with kali 2021.3. But after I installed kali 2021.4, the go install command worked without errors but nuclei wasn't installed in the system. Then I tried using the command: 
git clone https://github.com/projectdiscovery/nuclei.git; \
cd nuclei/v2/cmd/nuclei; \
go build; \
mv nuclei /usr/local/bin/; \
nuclei -version;

And this time it worked.

**Comment by notmarshmllow** [comment]
This is mainly due to `PATH` not set properly. After you install the packages using `go install ...` they will be located in a `/go/bin/` folder. In my case, this `go` folder was located in my home directory i.e. at `~/go/bin/`. So all I did was added this path to my `~/.profile` and it worked.

Add this line to your `~/.profile`: 
```bash
export PATH=$PATH:~/go/bin
```
Next run this command: `source ~/.profile`

Let me know if this works!
